packages/manager/src/managers/telemetry/TelemetryManager.ts

```diff
@@ -41,39 +41,47 @@ type TelemetryManagerContext = {
 };
 
 function assertTelemetryInitialized(
-	segmentClient: SegmentClient | undefined,
+	segmentClient: (() => Analytics) | undefined,
 ): asserts segmentClient is NonNullable<typeof segmentClient> {
-	if (segmentClient == undefined) {
+	if (segmentClient === undefined) {
 		throw new Error(
 			"Telemetry has not been initialized. Run `SliceMachineManager.telemetry.prototype.initTelemetry()` before re-calling this method.",
 		);
 	}
 }
 
 export class TelemetryManager extends BaseManager {
-	private _segmentClient: SegmentClient | undefined = undefined;
+	private _segmentClient: (() => Analytics) | undefined = undefined;
 	private _anonymousID: string | undefined = undefined;
 	private _userID: string | undefined = undefined;
 	private _context: TelemetryManagerContext | undefined = undefined;
 
 	async initTelemetry(args: TelemetryManagerInitTelemetryArgs): Promise<void> {
-		if (this._segmentClient) {
-			// Prevent subsequent initializations.
-			return;
-		}
+		const isTelemetryEnabled = await this.checkIsTelemetryEnabled();
+
+		this._segmentClient = () => {
+			const analytics = new Analytics({
+				writeKey: API_TOKENS.SegmentKey,
+				// Since it's a local app, we do not benefit from event batching the way a server would normally do, all tracking event will be awaited.
+				maxEventsInBatch: 1,
+				// TODO: Verify that this actually does not send data to Segment when false.
+				disable: isTelemetryEnabled === false,
```

Comment:
In Slice Machine UI we even have a rule to prevent that usage, right?

I started since I'm at Prismic to do explicit comparison, so I guess we should fix how we want to do, no?